gltfpack fails to write output data to tmp folder when compiling latest

I've been working on adding gltfpack as a tool in vcpkg. I've gotten v0.18 up there already but without basisu support for starters. Going back to add basisu support I've run into a little snag; I get a compilation error in meshopt's basisenc.cpp file: 

`basisenc.cpp(164): error C2661: 'basisu::job_pool::add_job': no overloaded function takes 2 arguments`
The code at commit e62c41e47cfdaf014c1e2a98dfeb571dd60c271f passes nullptr as a second parameter to `add_job`. But I can't find any commit in https://github.com/zeux/basis_universal where `add_job` accepts two parameters. Looks like v0.18 of gltfpack was released Jul 31, 2022. The closest commit in zeux/basis_universal looks to be 9009cae5407ef54f5030c1d0fabb1c2919c418bb from Jul 9, 2022. Is there some other patch that was applied to get this to compile? 

I have tried to work around this by just using the latest commits from both repos, which does compile, but I've run into a runtime error. When trying to compress a couple different glbs that I have with `-tc` I get the following outputs:
```
ERROR: Failed writing output data to file "<appdata>\Local\Temp\gltfpack-a35556.ktx2"
Warning: unable to encode image 0 (?), skipping (error encoding image)
```
I *don't* get these issues when using the v0.18 release of gltfpack downloaded off github. But I've tried compiling the latest build on Windows with LLVM and VS 2022 on Windows 11 with no success. 

So I guess this issue is two part: 
* Is there something I'm missing to compile the same v0.18 version of gltfpack?
* Am I somehow messing up compilation of latest that's causing this runtime failure?

For reference these have been my cmake configurations.
Clang 15.0.3 is set as my default compiler but the VS config uses MSVC.
```
cmake -S meshoptimizer\ -B build_llvm -G"Ninja Multi-Config" -DMESHOPT_BUILD_GLTFPACK=ON -DMESHOPT_BASISU_PATH=C:/Users/Arsine/Git/gltfpack_test/basis_universal

cmake -S meshoptimizer\ -B build -G "Visual Studio 17 2022" -DMESHOPT_BUILD_GLTFPACK=ON -DMESHOPT_BASISU_PATH=C:/Users/Arsine/Git/gltfpack_test/basis_universal
```
I dug in with a debugger and it seems like this is an error that comes from `basisu_comp.cpp` when `write_vec_to_file` fails. I've seen something kind of like this when contributing to KTX-Software where there is a bug in the `msvcrt` on large file writes https://github.com/KhronosGroup/KTX-Software/pull/574 but I haven't had time yet to test explicitly linking against the `ucrt` instead.

Am I just doing something wrong with the build system here? Should I not be using your fork of basisu to compile anymore? 

My test glbs are too big to upload here but one of them is just a glb version of Sponza. I can upload it somewhere else and link it if that's helpful.

--- a/gltf/basisenc.cpp
+++ b/gltf/basisenc.cpp
@@ -4,6 +4,7 @@
 
 #ifdef __clang__
 #pragma GCC diagnostic ignored "-Wunknown-warning-option"
+#pragma GCC diagnostic ignored "-Warray-bounds"
 #endif
 
 #ifdef __GNUC__
@@ -92,7 +93,7 @@ static void fillParams(basisu::basis_compressor_params& params, const char* inpu
 	params.m_status_output = false;
 }
 
-static const char* prepareEncode(basisu::basis_compressor_params& params, const cgltf_image& image, const char* input_path, const ImageInfo& info, const Settings& settings, TempFile& temp_input, TempFile& temp_output)
+static const char* prepareEncode(basisu::basis_compressor_params& params, const cgltf_image& image, const char* input_path, const ImageInfo& info, const Settings& settings, const std::string& temp_prefix, std::string& temp_input, std::string& temp_output)
 {
 	std::string img_data;
 	std::string mime_type;
@@ -107,10 +108,10 @@ static const char* prepareEncode(basisu::basis_compressor_params& params, const 
 
 	adjustDimensions(width, height, settings);
 
-	temp_input.create(mimeExtension(mime_type.c_str()));
-	temp_output.create(".ktx2");
+	temp_input = temp_prefix + mimeExtension(mime_type.c_str());
+	temp_output = temp_prefix + ".ktx2";
 
-	if (!writeFile(temp_input.path.c_str(), img_data))
+	if (!writeFile(temp_input.c_str(), img_data))
 		return "error writing temporary file";
 
 	int quality = settings.texture_quality[info.kind];
@@ -118,7 +119,7 @@ static const char* prepareEncode(basisu::basis_compressor_params& params, const 
 
 	const BasisSettings& bs = kBasisSettings[quality - 1];
 
-	fillParams(params, temp_input.path.c_str(),temp_output.path.c_str(), uastc, width, height, bs, info, settings);
+	fillParams(params, temp_input.c_str(), temp_output.c_str(), uastc, width, height, bs, info, settings);
 
 	return nullptr;
 }
@@ -130,8 +131,10 @@ void encodeImages(std::string* encoded, const cgltf_data* data, const std::vecto
 	basisu::vector<basisu::basis_compressor_params> params(data->images_count);
 	basisu::vector<basisu::parallel_results> results(data->images_count);
 
-	std::vector<TempFile> temp_inputs(data->images_count);
-	std::vector<TempFile> temp_outputs(data->images_count);
+	std::string temp_prefix = getTempPrefix();
+
+	std::vector<std::string> temp_inputs(data->images_count);
+	std::vector<std::string> temp_outputs(data->images_count);
 
 	for (size_t i = 0; i < data->images_count; ++i)
 	{
@@ -141,7 +144,7 @@ void encodeImages(std::string* encoded, const cgltf_data* data, const std::vecto
 		if (settings.texture_mode[info.kind] == TextureMode_Raw)
 			continue;
 
-		if (const char* error = prepareEncode(params[i], image, input_path, info, settings, temp_inputs[i], temp_outputs[i]))
+		if (const char* error = prepareEncode(params[i], image, input_path, info, settings, temp_prefix + "-" + std::to_string(i), temp_inputs[i], temp_outputs[i]))
 			encoded[i] = error;
 
 		// image is ready to encode in parallel
@@ -159,8 +162,16 @@ void encodeImages(std::string* encoded, const cgltf_data* data, const std::vecto
 			encoded[i] = "error decoding source image";
 		else if (results[i].m_error_code != basisu::basis_compressor::cECSuccess)
 			encoded[i] = "error encoding image";
-		else if (!readFile(temp_outputs[i].path.c_str(), encoded[i]))
+		else if (!readFile(temp_outputs[i].c_str(), encoded[i]))
 			encoded[i] = "error reading temporary file";
+	}
+
+	for (size_t i = 0; i < data->images_count; ++i)
+	{
+		if (!temp_inputs[i].empty())
+			removeFile(temp_inputs[i].c_str());
+		if (!temp_outputs[i].empty())
+			removeFile(temp_outputs[i].c_str());
 	}
 }
 #endif

--- a/gltf/fileio.cpp
+++ b/gltf/fileio.cpp
@@ -6,55 +6,25 @@
 #include <string.h>
 
 #ifdef _WIN32
-#include <io.h>
+#include <process.h>
 #else
 #include <unistd.h>
 #endif
 
-TempFile::TempFile()
-	: fd(-1)
+std::string getTempPrefix()
 {
-}
-
-TempFile::TempFile(const char* suffix)
-    : fd(-1)
-{
-	create(suffix);
-}
-
-TempFile::~TempFile()
-{
-	if (!path.empty())
-		remove(path.c_str());
-
-#ifndef _WIN32
-	if (fd >= 0)
-		close(fd);
-#endif
-}
-
-void TempFile::create(const char* suffix)
-{
-	assert(fd < 0 && path.empty());
-
 #if defined(_WIN32)
 	const char* temp_dir = getenv("TEMP");
-	path = temp_dir ? temp_dir : ".";
-	path += "\\gltfpack-XXXXXX";
-	(void)_mktemp(&path[0]);
-	path += suffix;
+	std::string path = temp_dir ? temp_dir : ".";
+	path += "\\gltfpack-temp";
+	path += std::to_string(_getpid());
+	return path;
 #elif defined(__wasi__)
-	static int id = 0;
-	char ids[16];
-	snprintf(ids, sizeof(ids), "%d", id++);
-
-	path = "gltfpack-temp-";
-	path += ids;
-	path += suffix;
+	return "gltfpack-temp";
 #else
-	path = "/tmp/gltfpack-XXXXXX";
-	path += suffix;
-	fd = mkstemps(&path[0], strlen(suffix));
+	std::string path = "/tmp/gltfpack-temp";
+	path += std::to_string(getpid());
+	return path;
 #endif
 }
 
@@ -137,4 +107,9 @@ bool writeFile(const char* path, const std::string& data)
 	int rc = fclose(file);
 
 	return rc == 0 && result == data.size();
+}
+
+void removeFile(const char* path)
+{
+	remove(path);
 }

--- a/gltf/gltfpack.h
+++ b/gltf/gltfpack.h
@@ -265,17 +265,7 @@ struct BufferView
 	size_t bytes;
 };
 
-struct TempFile
-{
-	std::string path;
-	int fd;
-
-	TempFile();
-	TempFile(const char* suffix);
-	~TempFile();
-
-	void create(const char* suffix);
-};
+std::string getTempPrefix();
 
 std::string getFullPath(const char* path, const char* base_path);
 std::string getFileName(const char* path);
@@ -283,6 +273,7 @@ std::string getExtension(const char* path);
 
 bool readFile(const char* path, std::string& data);
 bool writeFile(const char* path, const std::string& data);
+void removeFile(const char* path);
 
 cgltf_data* parseObj(const char* path, std::vector<Mesh>& meshes, const char** error);
 cgltf_data* parseGltf(const char* path, std::vector<Mesh>& meshes, std::vector<Animation>& animations, const char** error);

